File: 진우/12weeks/1934. 최소공배수/최소공배수.cc
#include <iostream>

using namespace std;

int main() {
    int T;
    cin >> T;   // 테스트 케이스 개수 입력

    int A, B;
    for (int i = 0; i < T; i++) {
        cin >> A >> B;  // 두 자연수 A와 B 입력

        // A와 B의 최소공배수(LCM)를 구하는 방식
        // 더 작은 수의 배수를 하나씩 증가시키며 체크

        if (A <= B) {
            int i = 1;
            while (true) {
                int result = 0;

                // B의 배수(B * i)가 A로 나누어떨어지면 최소공배수
                if ((B * i) % A == 0) {
                    result = B * i;
                    cout << result << "\n";
                    break;  // 찾았으므로 반복 종료
                }

                i++;  // 배수 증가
            }
        }
        else { // A > B일 때
            int i = 1;
            while (true) {
                int result = 0;

                // A의 배수(A * i)가 B로 나누어떨어지면 최소공배수
                if ((A * i) % B == 0) {
                    result = A * i;
                    cout << result << "\n";
                    break;  // 찾았으므로 반복 종료
                }
                i++;  // 배수 증가
            }
        }
    }

    return 0;
}

File: 진우/12weeks/1934. 최소공배수/최소공배수.py
# 유클리드 호제법
# 두 정수 a, b의 최대공약수를 반환한다.
def gcd(a, b):
    while b != 0:            # b가 0이 될 때까지 반복
        a, b = b, a % b      # a ← b, b ← a % b (유클리드 호제법 핵심)
    return a                 # b가 0이 되면 a가 최대공약수

T = int(input())             # 테스트 케이스 개수 입력

for _ in range(T):
    A, B = map(int, input().split())   # 두 자연수 A, B 입력

    if A <= B:
        g = gcd(A, B)                  # 두 수의 최대공약수 계산
        result = (A * B) // g          # 최소공배수 계산: LCM = A * B / GCD
        print(result)
    else:
        g = gcd(A, B)
        result = (A * B) // g
        print(result)
